core/config.h

```diff
@@ -58,6 +74,46 @@ class FuzzerConfig {
     [[nodiscard]] uint64_t getMinUpdateTimeInMicroseconds() const {
         return minUpdateTimeInMicroseconds;
     }
+
+    /// @brief Cast the TOML node to a specific type of value (encapsulated in `std::optional`).
+    template <typename T>
+    static std::optional<T> castTOMLNode(const toml::v3::node_view<const toml::v3::node> &node) {
+        if constexpr (std::is_same_v<T, int>) {
+            if (node.is_integer()) {
+                return node.as_integer()->get();
+            } else {
+                return std::nullopt;
+            }
+        } else if constexpr (std::is_same_v<T, uint64_t>) {
+            if (node.is_integer()) {
+                return node.as_integer()->get();
+            } else {
+                return std::nullopt;
+            }
+        } else if constexpr (std::is_same_v<T, size_t>) {
+            if (node.is_integer()) {
+                return node.as_integer()->get();
+            } else {
+                return std::nullopt;
+            }
+        } else if constexpr (std::is_same_v<T, std::vector<std::string>>) {
```

Comment:
You have a vector implementation but no std::string implementation? 
What are all the supported TOML casts?

Reply:
Currently, the supported TOML casts conform to the existing fuzzer configuration parameters (https://github.com/zzmic/rtsmith/blob/f4bfce79974162f86dc49f2de9889a23a1649457/core/config.h#L16-L31). Since there is no configuration parameter of type `std::string` alone, I have only implemented the casting of a vector of strings.

Reply:
[Here](https://marzer.github.io/tomlplusplus/classtoml_1_1table.html#type-casts) are the available TOML casts supported by TOML++.

Reply:
std::string is a base type so I would definitely add it. This function should only handle basetypes, the array could be separate function.